lib/library.rb

```diff
@@ -0,0 +1,68 @@
+class Library
+
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @book_checkout_count = {}
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      @books << book
+    end
+  end
+
+  def publication_time_frame_for(author)
+    hash = {start: first_published_book(author), end: last_published_book(author)}
+    hash
+  end
+
+  def last_published_book(author)
+    book_last = 0
+    author.books.each do |book|
+      if book.publication_year.to_i > book_last
+        book_last = book.publication_year.to_i
+      end
+    end
+    book_last.to_s
+  end
+
+  def first_published_book(author)
+    book_first = 2019
+    author.books.each do |book|
+      if book.publication_year.to_i < book_first
+        book_first = book.publication_year.to_i
+      end
+    end
+    book_first.to_s
+  end
```

Comment:
I'd be more impressed with helper methods if they were tested ;) #aaawtf
also, hard-coding 2019 in here will work just fine ... for another few months